autovalet_gazebo/scripts/autovalet_system.py

```diff
@@ -0,0 +1,176 @@
+                                "map", #source frame
+                                rospy.Time(0), #get the tf at first available time
+                                rospy.Duration(1.0)) #timeout after 1
+
+    current_pose = PoseStamped()
+    current_pose.header = transform.header
+    current_pose.pose.position.x = transform.transform.translation.x
+    current_pose.pose.position.y = transform.transform.translation.y
+    current_pose.pose.position.z = transform.transform.translation.z
+    current_pose.pose.orientation = transform.transform.rotation
+
+    return current_pose
+
+def arePosesClose(pose1, pose2):
+    result = True
+
+    if result and pose1.pose.position.x != pose2.pose.position.x:
+        result = False
+    if result and pose1.pose.position.y != pose2.pose.position.y:
+        result = False
+
+    return result
+
+class State:
+    Waiting = 0
+    Planning = 1
+    Executing = 2
+    Fail = 3
+    Success = 4
+
+class MoveBaseListener:
+    def __init__(self):
+        self.goal_sub       = rospy.Subscriber("/move_base_simple/goal", PoseStamped, self.goalCallback)
+        self.status_sub     = rospy.Subscriber("/move_base/status", GoalStatusArray, self.MBstatusCallback)
+        
+        self.goal_pose      = PoseStamped()
+        self.start_pose     = PoseStamped()
+
+        self.plan_state     = State.Waiting
+
+    def printState(self):
+        rospy.logwarn("---------")
+        output = "Current State: "
+
+        if self.plan_state == State.Waiting:
+            output += "WAITING FOR NEXT GOAL"
+        elif self.plan_state == State.Planning:
+            output += "PLANNING PATH TO GOAL"
+        elif self.plan_state == State.Executing:
+            output += "EXECUTING PLAN"
+        elif self.plan_state == State.Fail:
+            output += "FAILED TO FIND A PLAN"
+        elif self.plan_state == State.Success:
+            output += "SUCCESSFULLY REACHED GOAL"
+
+        rospy.logwarn(output)
+
+    def goalCallback(self,goal_msg):
+        
+
+        self.plan_state = State.Planning
+
+        self.goal_pose = goal_msg
+
+        self.start_pose = getCurrentPose()
+
+        self.printState()
+
+        rospy.logwarn("Goal:")
+        rospy.logwarn("Position")
+        rospy.logwarn("  x: %f",self.goal_pose.pose.position.x)
+        rospy.logwarn("  y: %f",self.goal_pose.pose.position.y)
+        rospy.logwarn("  z: %f",self.goal_pose.pose.position.z)
+        rospy.logwarn("Orientation");
+        rospy.logwarn("  x: %f",self.goal_pose.pose.orientation.x)
+        rospy.logwarn("  y: %f",self.goal_pose.pose.orientation.y)
+        rospy.logwarn("  z: %f",self.goal_pose.pose.orientation.z)
+        rospy.logwarn("  w: %f",self.goal_pose.pose.orientation.w)
+
+    def MBstatusCallback(self,status_array_msg):
+
+        # print(status_array_msg.__dict__)
+        ACTIVE = 1
+        SUCCEEDED = 3
+        ABORTED = 4
+
+        if status_array_msg.status_list:
```

Comment:
can you comment a bit what those conditions are in general? I found it hard to keep track of what is what. maybe just a high level concept would be enough.